sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,431 @@
+
+        resource.Filter = {
+          CorrelationId: correlationFilter.correlationId,
+          Label: correlationFilter.label,
+          To: correlationFilter.to,
+          ReplyTo: correlationFilter.replyTo,
+          ReplyToSessionId: correlationFilter.replyToSessionId,
+          ContentType: correlationFilter.contentType,
+          SessionId: correlationFilter.sessionId,
+          MessageId: correlationFilter.messageId,
+          Properties: correlationFilter.userProperties
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "CorrelationFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      }
+    }
+
+    if (rule.action == undefined || rule.action.sqlExpression == undefined) {
+      // Defaults to creating an empty rule action instance if none specified
+      resource.Action = {};
+      resource.Action[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "EmptyRuleAction",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      resource.Action = {
+        SqlExpression: rule.action.sqlExpression,
+        Parameters: getRawSqlParameters(rule.action.sqlParameters),
+        CompatibilityLevel: 20,
+        RequiresPreprocessing: getStringOrUndefined(rule.action.requiresPreprocessing)
+      };
+      resource.Action[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "SqlRuleAction",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    }
+
+    return serializeToAtomXmlRequest("RuleDescription", resource);
+  }
+
+  async deserialize(response: HttpOperationResponse): Promise<HttpOperationResponse> {
+    return deserializeAtomXmlResponse(["TopicName", "SubscriptionName", "RuleName"], response);
+  }
+}
+
+enum SqlParameterType {
+  Integer = "l28:int",
+  String = "l28:string",
+  Long = "l28:long",
+  Date = "l28:date"
+}
+/**
+ * Represents type of SQL `Parameter` in ATOM based management operations
+ */
+export type SqlParameter = {
+  key: string;
+  value: string | number;
+  type: string;
+};
+
+/**
+ * Internal representation of SQL parameter info
+ */
+type RawSqlParameter = {
+  Key: string;
+  Value: any;
+};
+
+/**
+ *  @ignore
+ * Helper utility to retrieve array of `SqlParameter` from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getSqlParametersOrUndefined(value: any): SqlParameter[] | undefined {
+  const parameters: SqlParameter[] = [];
+
+  // Ignore special case as Service Bus treats "" as a valid value for SQL parameters
+  if (typeof value === "string" && value.trim() === "") {
+    return undefined;
+  }
+
+  const jsObject: any = getJSObjectOrUndefined(value);
+  if (jsObject == undefined) {
+    return undefined;
+  }
+
+  try {
+    let rawParameters = jsObject["KeyValueOfstringanyType"];
+    if (rawParameters && rawParameters.length && rawParameters.length > 0) {
```

Comment:
A check on length > 0 is redundant when the length itself is checked
```suggestion
    if (rawParameters && rawParameters.length) {
```